documents/typescript-guidelines.md

```diff
@@ -0,0 +1,134 @@
+# TS Conventions in Deriv-com (Work-in-progress)
+
+Good reads:
+
+-   https://www.sitepoint.com/react-with-typescript-best-practices/
+-   https://onesignal.com/blog/effective-typescript-for-react-applications/
```

Comment:
```suggestion
```
We can remove this one as it has some points that is contradicatory to what we have here to avoid confusion.